### PR TITLE
20260406 bond risk [GEN-7371]

### PR DESCRIPTION
**Pull request by kronael**
- simplify to fix a bug
- add some Claude-Generated edge-case tests

### CODE REVIEW

File: packages/ds-sam-sdk/src/calculations.ts [hunk 1]
```diff
@@ -131,24 +130,16 @@ export type BondRiskFeeResult = {
 }
 
 /**
+ * Checks if a validator's claimable bond covers its exposed
+ * stake (bond minus unprotected reserve vs projected exposed
+ * stake times minBondPmpe). If underfunded, computes forced
```
Comment (kronael) [inline packages/ds-sam-sdk/src/calculations.ts]:
```suggestion
 * stake. If underfunded, computes forced
```

File: packages/ds-sam-sdk/test/sam.test.ts [hunk 2]
```diff
@@ -180,73 +186,6 @@ describe('sam', () => {
         expect(isFinite(revShare.auctionEffectiveBidPmpe)).toBe(true),
       )
 
-      result.auctionData.validators
```
Comment (kronael) [inline packages/ds-sam-sdk/test/sam.test.ts:183]: That test was an exact duplicate of run() assigns effective participating bids — same setup (60 validators, same builder params), same assertions. The only difference was the first test had
   a duplicated filter block (lines 281-285 repeated 275-279). The "final only" test was identical to the first one minus that repeated block.

  I removed the duplicate test and the repeated filter block, keeping the first test with all unique assertions intact.

**Review by ochaloup — DISMISSED** [review]
Fine to me.

* `ARCHITECTURE.md`  states `BOND: Protected stake cap = claimableBondBalanceSol / (bondPmpe / 1000)` but the code `bondStakeCapSam()` uses `validator.bondBalanceSol`

**Review by ochaloup — APPROVED** [review]
